metrics/prometheus/factory.go

```diff
@@ -174,14 +174,42 @@ func (f *Factory) Timer(options metrics.Options) metrics.Timer {
 	opts := prometheus.HistogramOpts{
 		Name:    name,
 		Help:    help,
-		Buckets: f.buckets,
+		Buckets: asFloatBuckets(options.Buckets),
 	}
 	hv := f.cache.getOrMakeHistogramVec(opts, labelNames)
 	return &timer{
 		histogram: hv.WithLabelValues(f.tagsAsLabelValues(labelNames, tags)...),
 	}
 }
 
+func asFloatBuckets(buckets []time.Duration) []float64 {
+	data := make([]float64, len(buckets))
+	for i := range data {
+		data[i] = float64(buckets[i])
```

Comment:
doesn't Prometheus client expects `double(1.0) == 1sec`?  Do we need:
```
data[i] =  float64(buckets[i]) / float64(time.Second)
```
?

Reply:
Possibly for Timer (duration) buckets - but for Histogram this is just a list of floats,

Reply:
but this function is used for Timer buckets, is it not?  Since it accepts `[]time.Duration`. I think you need to divide this by time.Second, since this is what the Record function of the timer does:

https://github.com/jaegertracing/jaeger-lib/blob/3fa9aa04b65a266a1bbe0a5c8ac1408281b2a64a/metrics/prometheus/factory.go#L214-L216

Reply:
Yep sorry - not sure what I was thinking.